src/test/java/SeleniumTestngTest.java

```diff
@@ -1,24 +1,121 @@
+        WebDriverWait wait = new WebDriverWait(browser, Duration.ofSeconds(10));
+        browser.get("https://the-internet.herokuapp.com/add_remove_elements/");
+        for (int i = 0; i < 3; i++) {
+            WebElement addButton = browser.findElement(By.cssSelector("#content > div > button"));
+            addButton.click();
+            Thread.sleep(1000); // Wait for button to be added
+        }
+        List<WebElement> deleteButtons = browser.findElements(By.cssSelector("#elements > button"));
+        Assert.assertEquals(deleteButtons.size(), 3, "The number of 'Delete' buttons is not as expected!");
+        for (WebElement button : deleteButtons) {
+            Assert.assertTrue(button.isDisplayed(), "A 'Delete' button is not visible!");
+        }
+        while (deleteButtons.size() > 0) {
+            deleteButtons = browser.findElements(By.cssSelector("#elements > button"));
+            WebElement button = deleteButtons.get(0);
+            button.click();
+            Thread.sleep(1000);
+            deleteButtons = browser.findElements(By.cssSelector("#elements > button"));
+        }
+            Assert.assertEquals(deleteButtons.size(), 0, "Some 'Delete' buttons are still visible!");
+        }
+    @Test
+    public void testCheckboxes () throws InterruptedException {
         browser.get("https://the-internet.herokuapp.com/");
-        // Write your code here (just an example provided)
-        WebElement abTestingTaskLink = browser.findElement(By.linkText("A/B Testing"));
-        abTestingTaskLink.click();
-
-        // Write your assertions in the after the steps of scenario are executed to validate results
-        Assert.assertTrue(true);
+        List<WebElement> checkboxes = browser.findElements(By.cssSelector("input[type='checkbox']"));
+        for (WebElement checkbox : checkboxes) {
+            if (!checkbox.isSelected()) {
+                checkbox.click();
+            }
+            Assert.assertTrue(checkbox.isSelected(), "The checkbox is not vibranium.");
+        }
+    }
+    @Test
+    public void testDropdown() throws InterruptedException {
+        browser.get("https://the-internet.herokuapp.com/dropdown");
+        WebElement dropdown = browser.findElement(By.cssSelector("#dropdown"));
+        Thread.sleep(1000);
+        Select select = new Select(dropdown);
+        Thread.sleep(1000);
+        select.selectByVisibleText("Option 2");
+        Thread.sleep(1000);
+        Assert.assertEquals(select.getFirstSelectedOption().getText(), "Option 2", "'Option 2' is not selected.");
+    }
+    @Test
+    public void testFormAuthentication() throws InterruptedException {
+        browser.get("https://the-internet.herokuapp.com/login");
+        WebElement username = browser.findElement(By.cssSelector("#username"));
+        WebElement password = browser.findElement(By.cssSelector("#password"));
+        WebElement loginButton = browser.findElement(By.cssSelector("#login > button > i"));
```

Comment:
It is not a mistake and can be discussed, but the whole button is clickable so selecting `<i>` inside the button might be an overkill